### PR TITLE
 Task #13 Method toString() in Bowling class

### PR DESCRIPTION
**Pull request by szymonjanas**
Method toString() to show data like Lane number, players names and scores.
There is needed implementation method to show current Game Status.

### CODE REVIEW

File: src/Bowling.cpp [hunk 1]
```diff
@@ -64,3 +65,37 @@ void Bowling::readDirectory(path p)
...
+
+std::string Bowling::toString() const
+{
+    std::stringstream ss;
+    std::string data;
+    for (int i = 0; i < getNumberOfLanes(); i++)
+    {
+        Lane lane = getLane(0);
+        std::string laneData;
+        std::stringstream no;
```
Comment (Angan7a) [inline src/Bowling.cpp]: The name `no` says nothing. Maybe better name will be `laneNumber`.

File: src/Bowling.cpp [hunk 2]
```diff
@@ -64,3 +65,37 @@ void Bowling::readDirectory(path p)
...
+std::string Bowling::toString() const
+{
+    std::stringstream ss;
+    std::string data;
+    for (int i = 0; i < getNumberOfLanes(); i++)
+    {
+        Lane lane = getLane(0);
+        std::string laneData;
+        std::stringstream no; 
+        no << ++i;
```
Comment (Angan7a) [inline src/Bowling.cpp]: In my opinion you want do `(i+1)`, not increment variable which is used in for loop.

File: src/Bowling.cpp [hunk 3]
```diff
@@ -64,3 +65,37 @@ void Bowling::readDirectory(path p)
...
+        std::string numberOfLane = no.str();
+            laneData.insert(laneData.size(), "### Lane ");
+            laneData.insert(laneData.size(), numberOfLane);
+            laneData.insert(laneData.size(), ": ");
+            laneData.insert(laneData.size(), "GAME STATUS"); // game status method is needed
+            laneData.insert(laneData.size(), " ###\n");
```
Comment (Angan7a) [inline src/Bowling.cpp]: Maybe better solution will be `ss << "### Lane " << (i+1) << ": " << "GAME STATUS" << " ###\n"`

File: src/Bowling.cpp [hunk 4]
```diff
@@ -64,3 +65,37 @@ void Bowling::readDirectory(path p)
...
+        std::string namesAndScore;
+        for (int i = 0; i < lane.getNumberOfGames(); i++)
```
Comment (Angan7a) [inline src/Bowling.cpp]: If you use a loop in a loop, use different variable names, please.

File: src/Bowling.cpp [hunk 5]
```diff
@@ -64,3 +65,37 @@ void Bowling::readDirectory(path p)
...
+        {
+            std::stringstream scr;
+            scr << lane.getGame(i).score();
+            std::string score = scr.str();
+            namesAndScore.insert(namesAndScore.size(), lane.getGame(i).getName());
+            namesAndScore.insert(namesAndScore.size(), " ");
+            namesAndScore.insert(namesAndScore.size(), score);
+            namesAndScore.insert(namesAndScore.size(), "\n");
```
Comment (Angan7a) [inline src/Bowling.cpp]: Maybe better solution will be `ss << lane.getGame(j).getName() << " " << lane.getGame(j).score() << "/n";`

File: tests/BowlingTestsToStringMethod.cpp [hunk 6]
```diff
@@ -0,0 +1,102 @@
...
+    Game game6("Name6:7|X|X|X");
+    lane2.addGame(game4);
+    lane2.addGame(game5);
+    lane2.addGame(game6);
+    bowling.addLane(lane2);
+    Lane lane3;
+    Game game7("Name7:7|9|X|X|6/|X|X|6|7|8");
+    Game game8("Name8:7|9|X|X|8|9|3|6|7|8");
+    Game game9("Name9:7|X|X|X|6/|8|3|6|X|8");
+    Game game10("Name10:X|X|X|X|X|X|X|X|X|X|X|X");
```
Comment (Tomek52) [inline tests/BowlingTestsToStringMethod.cpp]: @szymonjanas If you have strike and bonus balls game should look like `X|X|X|X|X|X|X|X|X|X||XX`

**Pull request merged**